quickcheck/Test/Tasty/QuickCheck.hs

```diff
@@ -199,20 +202,18 @@ instance IsTest QC where
     , Option (Proxy :: Proxy QuickCheckMaxShrinks)
     ]
 
-  run opts (QC prop) _yieldProgress = do
+  run opts (QC prop) yieldProgress = do
     (replaySeed, args) <- optionSetToArgs opts
-
     let
       QuickCheckShowReplay showReplay = lookupOption opts
       QuickCheckVerbose    verbose    = lookupOption opts
       maxSize = QC.maxSize args
-      testRunner = if verbose
-                     then QC.verboseCheckWithResult
-                     else QC.quickCheckWithResult
       replayMsg = makeReplayMsg replaySeed maxSize
 
     -- Quickcheck already catches exceptions, no need to do it here.
-    r <- testRunner args prop
+    r <- quickCheck yieldProgress
```

Comment:
30-40% penalty for displaying progress is not quite acceptable in my opinion. Can we update the output only when `progressPercent` changes?

Reply:
`ppProgressOrResult` does the caching, so  we only do `IO` when the value changes.  I experimented with updating the output every now and then (e.g. <=100ms) and it didn't improve the speed, which suggest the inefficiency is elsewhere.

Reply:
This suggest that the slow down is happening on QuickCheck side: indeed `Test.QuickCheck.Text.putTemp` is very inefficient. 

The callback we pass to `newTerminal` has type `String -> IO ()`. What if we ignore completely `String` (hopefully, it will not be constructed at all = no slowdown) and just bump our own `IORef` counter?

Reply:
I've raised https://github.com/nick8325/quickcheck/pull/357 to improve `Test.QuickCheck.Text.putTemp`.

Reply:
`QuickCheck-2.14.3` has been released. @coot could you possibly please give another go to your benchmarks?

Reply:
This branch
```
./test -p "/QuickCheck/&&/Fermat's little theorem/" --quickcheck-tests=1000000
Tests
  Properties
    (checked by QuickCheck)
      Fermat's little theorem: OK (1.07s)
        +++ OK, passed 1000000 tests.
```
vs 9ff66c233 (the merge-base):
```
./test -p "/QuickCheck/&&/Fermat's little theorem/" --quickcheck-tests=1000000
Tests
  Properties
    (checked by QuickCheck)
      Fermat's little theorem: OK (0.98s)
        +++ OK, passed 1000000 tests.

All 1 tests passed (0.99s)
```

:tada: